externals/tecio/teciompisrc/DataSetWriterMPI.cpp

```diff
@@ -18,7 +18,7 @@ unsigned int oldOutputFormat = _set_output_format(_TWO_DIGIT_EXPONENT);
 if (!___2337->___4638(___904 + 1)) continue; if (___2337->zoneIsPartitioned(___904 + 1) || ___2337->zoneGetOwnerProcess(___904 + 1) == m_localProcess) { std::vector<___372> ___4564; ___372 ___4499; getZoneSharing(___4564, ___4499, ___904, ___341, szpltFile.___844()); zoneWriterMap[fileZone] = ___4710.___4708(*m_varIter, ___904, ___341, ___4564, ___4499); uint64_t zoneSize = zoneWriterMap[fileZone]->zoneFileSize(szpltFile.___2002() == ___4226); if (!___2337->zoneIsPartitioned(___904 + 1)) { communicationCollection.sendScalarCopy(zoneSize, m_mainProcess, TecioMPI::ZONE_FILE_SIZE); communicationCollection.receiveScalar(zoneFileLocMap[fileZone], m_mainProcess, TecioMPI::ZONE_FILE_LOC); } } } communicationCollection.___4446(); } m_zoneIter->reset(); ___4636 const ___341 = m_zoneIter->baseItem(); while (___2039 && m_zoneIter->hasNext()) { ___4636 const ___904 = m_zoneIter->next(); ___4636 const fileZone = ___904 - ___341; if (!___2337->___4638(___904 + 1)) continue; if (___2337->zoneIsPartitioned(___904 + 1) || ___2337->zoneGetOwnerProcess(___904 + 1) == m_localProcess) { ___2039 = zoneWriterMap[fileZone]->writeZone(szpltFile, zoneFileLocMap[fileZone]); if (m_localProcess == m_mainProcess) { szpltZoneHeaderFileLocs[fileZone] = zoneWriterMap[fileZone]->getZoneHeaderFilePosition(); m_varIter->reset(); ___4352 const baseVar = m_varIter->baseItem(); while (m_varIter->hasNext()) { ___4352 const datasetVar = m_varIter->next(); ___4352 const fileVar = datasetVar - baseVar; m_zoneVarMetadata->m_vzMinMaxes[fileVar][fileZone] = zoneWriterMap[fileZone]->varMinMax(datasetVar); } } else if (!___2337->zoneIsPartitioned(___904 + 1)) { MPINonBlockingCommunicationCollection communicationCollection(m_communicator); ___1393 headerFileLoc = zoneWriterMap[fileZone]->getZoneHeaderFilePosition(); communicationCollection.sendScalarCopy(headerFileLoc, m_mainProcess, TecioMPI::HEADER_FILE_LOC); std::vector<___2479> minMaxValues; minMaxValues.reserve(m_varIter->___2812()); m_varIter->reset(); while (m_varIter->hasNext()) { ___4352 const ___4336 = m_varIter->next(); minMaxValues.push_back(zoneWriterMap[fileZone]->varMinMax(___4336)); } communicationCollection.sendVectorCopy(minMaxValues, m_mainProcess, TecioMPI::VAR_MINMAX_VALUES_SIZE, TecioMPI::VAR_MINMAX_VALUES_VEC); communicationCollection.___4446(); } } else if (m_localProcess == m_mainProcess) { int zoneOwner = ___2337->zoneGetOwnerProcess(___904 + 1); MPICommunicator communicator(m_communicator); communicator.receiveScalar(szpltZoneHeaderFileLocs[fileZone], zoneOwner, TecioMPI::HEADER_FILE_LOC); SimpleVector<___2479> minMaxValues; communicator.receiveVector(minMaxValues, zoneOwner, TecioMPI::VAR_MINMAX_VALUES_SIZE, TecioMPI::VAR_MINMAX_VALUES_VEC); ___478(minMaxValues.size() == m_varIter->___2812()); m_varIter->reset(); ___4352 const baseVar = m_varIter->baseItem(); while (m_varIter->hasNext()) { ___4352 const fileVar = m_varIter->next() - baseVar; m_zoneVarMetadata->m_vzMinMaxes[fileVar][fileZone] = minMaxValues[fileVar];
```

Comment:
It might be a long code line  ... but at least it is straight forward to understand ... 🙈
Not scope of this PR though

Reply:
well, the tecio code is obfuscated ...

Reply:
Tuesday is SU2 funfact day:
The longest code line in our repo can be found in `externals/tecio/boost/phoenix/core/preprocessed/actor_50.hpp +9` 🥇 and is this long (take a guess)

<details><summary> click me </summary>
<p>

139.114 🎉 

</p>
</details>

Yeah I know boost is not written by anyone here but I was still a bit amazed

The longest line in /Common or SU2_CFD has in contrast a dissapointing 714 chars and can be found in `Common/include/datatype_structure.inl +173`

<details><summary>click for the used command</summary>
<p>

```bash
find . \( -name '*.cpp' -o -name '*.hpp' -o -name '*.inl' -o -name '*.h' -o -name '*.c' \) -type f -exec wc -L {} \; | tee wc-L.log &&  echo ""  && cut -d: -f 2 wc-L.log | sort -n | tail -n 1
```

</p>
</details>